mrbgems/mruby-regexp/test/string_index.rb

```
@@ -544,6 +578,45 @@ def fake.match(str, pos = 0); raise "must not be called"; end
+  # a group that did not take part leaves nil behind, and `$+` reaches past it
+  assert_equal 1, "abc".rindex(/(b)(z)?/)
+  assert_nil $2
+  assert_equal "b", $+
+
+  # and a search that finds nothing clears all of them, not `$~` alone
+  "zzz" =~ /(z)/
+  assert_nil "abc".rindex(/x/)
+  assert_nil $&
+  assert_nil $`
+  assert_nil $'
+  assert_nil $1
```

RuboCop (1.88.2)

[warning] $1 is out of range (no regexp capture groups detected).

(Lint/OutOfRangeRegexpRef)